src/Microsoft.Health.Fhir.Shared.Api/Utilities/CallerIpAddressRetriever.cs

```diff
@@ -0,0 +1,41 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Linq;
+using EnsureThat;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Health.Fhir.Core.Features.Context;
+using Microsoft.Health.Fhir.Core.Utilities;
+
+namespace Microsoft.Health.Fhir.Api.Utilities
+{
+    public class CallerIpAddressRetriever : ICallerIpAddressRetriever
+    {
+        private readonly IFhirRequestContextAccessor _fhirRequestContextAccessor;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+
+        public CallerIpAddressRetriever(IFhirRequestContextAccessor fhirRequestContextAccessor, IHttpContextAccessor httpContextAccessor)
+        {
+            EnsureArg.IsNotNull(fhirRequestContextAccessor, nameof(fhirRequestContextAccessor));
+            EnsureArg.IsNotNull(httpContextAccessor, nameof(httpContextAccessor));
+
+            _fhirRequestContextAccessor = fhirRequestContextAccessor;
+            _httpContextAccessor = httpContextAccessor;
+        }
+
+        public string CallerIpAddress
```

Comment:
Is this already a thing with `UseForwardedHeaders`?
- https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.builder.forwardedheadersextensions.useforwardedheaders?view=aspnetcore-2.2#Microsoft_AspNetCore_Builder_ForwardedHeadersExtensions_UseForwardedHeaders_Microsoft_AspNetCore_Builder_IApplicationBuilder_
- https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer?view=aspnetcore-2.2#troubleshoot

Reply:
:open_mouth: I didn't know it existed. That makes the code simpler.

Reply:
This was much easier to implement.